javascript-carousel/main.js

```diff
@@ -0,0 +1,44 @@
+var $image = document.querySelectorAll('.image');
+var $progressIndicators = document.querySelector('.progress-indicators');
+var $circles = document.querySelectorAll('.circle');
+var $leftArrow = document.querySelector('.left');
+var $rightArrow = document.querySelector('.right');
+var position = 0;
+
+var interval = setInterval(rotate, 3000);
+
+$progressIndicators.addEventListener('click', function (e) {
+  var view = e.target.dataset.view;
+  updateImage(view);
+});
+
+$rightArrow.addEventListener('click', rotate);
+
+$leftArrow.addEventListener('click', function () {
+  position--;
+  updateImage(position);
+  if (position === -1) {
+    updateImage('4');
+  }
+});
+
+function rotate() {
+  position++;
+  updateImage(position);
+  if (position === 5) {
+    updateImage('0');
+  }
+}
+
+function updateImage(currentValue) {
+  var i = parseInt(currentValue);
```

Comment:
It's best to avoid forcing a function to "fix" its own inputs. You can do `parseInt` on line 11 instead. That way you can stick with `i` as the index in your loop here and compare it to `currentValue` directly.